### ISSUE TITLE
Rule proposal: `jest/prefer-strict-mock-assertions`

### ISSUE THREAD

**Issue by nicolasgillot** [jest-community/eslint-plugin-jest#1820]
### Problem
Developers often use `toHaveBeenCalledWith()` without verifying the call count, leading to tests that pass when a mock is called multiple times unexpectedly.

### Solution
Require `toHaveBeenCalledTimes()` when using `toHaveBeenCalledWith()` to ensure complete mock validation.

### Examples

```javascript
// ❌ Potentially flaky - passes if called 1 + times
expect(mockFn).toHaveBeenCalledWith(expectedArg)

// ✅ Explicit and safe - ensures exact call count
expect(mockFn).toHaveBeenCalledTimes(1)
expect(mockFn).toHaveBeenCalledWith(expectedArg)
```

### Benefits
- Prevents false positives in tests
- Makes test intentions explicit
- Improves test reliability and maintainability
- Follows testing best practices

Are you willing to implement this rule? Yes

**Comment by G-Rath** [comment]
yeah I guess I could see this being useful - not sure about the name though, maybe something like `pair-to-have-been-called-assertions`?

**Comment by nicolasgillot** [comment]
Great point about the naming! `pair-to-have-been-called-assertions` is indeed more specific and descriptive.

It clearly indicates that the rule ensures **pairing** `toHaveBeenCalledWith()` with `toHaveBeenCalledTimes()`.

I'll go with that name for the implementation. Should I proceed with creating a PR?

**Comment by SimenB** [comment]
PR very much welcome, yeah 😀

**Comment by nicolasgillot** [comment]
Hello, I have opened the pull request [here](https://github.com/jest-community/eslint-plugin-jest/pull/1848) 🙂
